src/VisualStudio/CSharp/Test/PersistentStorage/TestCloudCacheServiceProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Storage;
+using Microsoft.CodeAnalysis.UnitTests.WorkspaceServices.Mocks;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.ServiceHub.Framework.Services;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Cache;
+using Microsoft.VisualStudio.Cache.SQLite;
+using Microsoft.VisualStudio.LanguageServices.Storage;
+using Microsoft.VisualStudio.RpcContracts.Caching;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.UnitTests.WorkspaceServices
+{
+    internal class TestCloudCacheServiceProvider : IRoslynCloudCacheServiceProvider
+    {
+        private readonly IThreadingContext _threadingContext;
+        private readonly string _relativePathBase;
+
+        public TestCloudCacheServiceProvider(IThreadingContext threadingContext, string relativePathBase)
+        {
+            Console.WriteLine($"Instantiated {nameof(TestCloudCacheServiceProvider)}");
+            _threadingContext = threadingContext;
+            _relativePathBase = relativePathBase;
+        }
+
+        public async ValueTask<IRoslynCloudCacheService> CreateCacheAsync(CancellationToken cancellationToken)
+        {
+            // Directly access VS' CacheService through their library and not as a brokered service. Then create our
+            // wrapper CloudCacheService directly on that instance.
+            var authorizationServiceClient = new AuthorizationServiceClient(new AuthorizationServiceMock());
+            var solutionService = new SolutionServiceMock();
+            var fileSystem = new FileSystemServiceMock();
+            var serviceBroker = new ServiceBrokerMock()
```

Comment:
The `IServiceBroker` implementation we use in testing OOP is `InProcRemoteHostCLient.InProcServiceBroker`. I'm not sure what we use for corresponding in-proc implementation, but ideally these merge.